src/cpp/rtps/transport/TCPTransportInterface.cpp

```diff
@@ -757,10 +757,44 @@ bool TCPTransportInterface::OpenOutputChannel(
             listening_port = config->listening_ports.front();
         }
 
+        bool local_lower_interface = false;
+        if (IPLocator::getPhysicalPort(physical_locator) == listening_port)
+        {
+            std::vector<Locator> list;
+            std::vector<fastrtps::rtps::IPFinder::info_IP> locNames;
+            get_ips(locNames);
+            for (const auto& infoIP : locNames)
+            {
+                Locator newloc(physical_locator);
+                if (transport_kind_ == LOCATOR_KIND_TCPv4)
+                {
+                    IPLocator::setIPv4(newloc, infoIP.locator);
+                }
+                else
+                {
+                    IPLocator::setIPv6(newloc, infoIP.locator);
+                }
+
+                if (is_interface_allowed(newloc))
+                {
+                    list.push_back(newloc);
+                }
```

Comment:
```suggestion
            std::vector<fastrtps::rtps::IPFinder::info_IP> local_interfaces;
            get_ips(local_interfaces);
            for (const auto& interface : local_interfaces)
            {
                Locator interface_loc(interface.locator);
                interface_loc.port = physical_locator.port;

                if (is_interface_allowed(interface_loc))
                {
                    list.push_back(interface_loc);
                }
```

Reply:
I have also set the locator.kind of the _interface_loc_ to make sure the comparison of locators only take into account the address.

Reply:
That should have been already done by `get_ips`